### ISSUE TITLE
Blank, no cursor, no working shortcuts.

### ISSUE THREAD

**Issue by ahmadraniri1994** [jbuchermn/newm#111]
I'm trying the latest of newm and getting blank appearance, no cursor, and no working shortcuts.
Log said about backlight stuffs and when I disable it, still no effect.
Log I got  : 
[newm_log.txt](https://github.com/jbuchermn/newm/files/8673888/newm_log.txt)

**Comment by ahmadraniri1994** [comment]
I launched it from TTY, other Wayland compositor and X window manager, still have same behaviours.

**Comment by ahmadraniri1994** [comment]
I'll try to install greeter and see what will happen.

**Comment by ahmadraniri1994** [comment]
> I'll try to install greeter and see what will happen.

Still same.

**Comment by jbuchermn** [comment]
Please also post your config. If you installed the latest master, you're on v0.3alpha. Did you use v0.3 branch before?

**Comment by ahmadraniri1994** [comment]
O, Sorry for that.
I just install it with `pip3 install git+https://github.com/jbuchermn/newm`
Here my [config](https://raw.githubusercontent.com/ahmadraniri1994/logs-and-notes/main/default_config.py) .
Note : I comment some lines.

**Comment by jbuchermn** [comment]
Can you try changing the path in 
```
background = {
    'path': os.path.dirname(os.path.realpath(__file__)) + '/resources/wallpaper.jpg',
    'anim': True
}
```
to something that certainly exists? This default config does not work in all cases. I just want to see if anything gets rendered.

Also please test
```
pywm = {
    'renderer_mode': 'wlr'
}
```
does this change anything?

**Comment by ahmadraniri1994** [comment]
Unfortunately, it's still same.
I try to change the background path to :
```
background = {
    'path': '/home/ahmad/somewp.jpg',
    'anim': True
}
```
And add :

```
pywm = {
    'renderer_mode': 'wlr'
}
```

But it gives no change. I'll probably stick to v0.2 .

**Comment by jbuchermn** [comment]
Okay, thanks. It appears the renderer is not the issue...

You don't set your keyboard layout in the config - could that be the reason for keychords not working? Obviously you won't see anything, but working keychords should at least show up in the log, if it starts alacritty e.g..

Does it work on v0.2?

**Comment by ahmadraniri1994** [comment]
> You don't set your keyboard layout in the config ?

I set it up and the shortcuts still don't work, cursor still gone.

Newm v0.2 works for me.

**Comment by jbuchermn** [comment]
Yeah this wouldn't change the rendering issues. But does anything show up in the logs if you press keybindings?

It's interesting that v0.2 works..

**Comment by ahmadraniri1994** [comment]
> But does anything show up in the logs if you press keybindings?

unfortunately nothing show up in the logs.

Update what I tried and got  on newm v0.3 :
- Background will show up when I change the `anim` to `False`.
```
background = {
    'path': os.path.dirname(os.path.realpath(__file__)) + '/resources/wallpaper.jpg',
    'anim': False
}
```
- Cursor still doesn't show.
- Keybinds still don't work though I set the keyboard layout : 
```
pywm = {
    "xkb_model": "pc104",
    "xkb_layout": "us",
}
```

I want to stick with v0.2 for now, btw, how to prevent high cpu usage when using newm v0.2 ? the fan spins and make noise when I'm using newm, it's not like the other wayland compositors.

**Comment by jbuchermn** [comment]
Can you post a current log with the config that shows Blythe background?

Regarding v0.2, what you can do is debug and submit a PR. I'm no longer going to actively work on old versions, but I'm always happy about contributions.

However, high cpu usage is not something that's ever popped up before, so I think there might be something bigger going wrong (possibly the same issue preventing rendering on v0.3...)

Does the behaviour change in any way if launched inside sway, or another compositor instead of from tty?

**Comment by lidgnulinux** [comment]
> Can you post a current log with the config that shows Blythe background?

sure, this is the [log](https://github.com/lidgnulinux/logforstuffs./blob/main/newm_log).

> Does the behaviour change in any way if launched inside sway, or another compositor instead of from tty?

Both (launch from tty and from other wayland compositor) give same behaviour (high cpu usage on v0.2, blank and no cursor for v0.3).
Pic (sometimes, it's almost 100% on one of the cpu, I don't know why) : 
![2022-05-24T05:13:08,777735462+07:00](https://user-images.githubusercontent.com/99301306/169914550-952cfeed-4715-4ada-ab0b-41c446dc8bbd.png)

**Comment by jbuchermn** [comment]
I haven't had time to take a detailed look, but from a quick glance at the screenshot it seems like cpu usage stems from newm_panel_basic instead of newm, right?

How about disabling it?

I'll take a look at your v0.3 log as soon as I have time - fixing the current version is my priority.

**Comment by pedrotrschneider** [comment]
I seem to be having the same issue. Reproduced exactly what was described. At first, no wallpaper, no mouse cursor and no keybindings were working. Tried adding
```python
pywm = {
    'renderer_mode': 'wlr'
}
```
to no avail. Disabling wallpaper animations made the wallpaper appear, but it remained unresponsive otherwise.
My config file and logs are attached (had to attach both as .txt because github, sorry for that).

[newm_log](https://github.com/jbuchermn/newm/files/9006730/newm_log.txt)
[config](https://github.com/jbuchermn/newm/files/9006734/config.txt)

Newm was installed from pip, and the distro is Fedora 36 with kernel 5.18 (if it matters at all). Newm was launched from the TTY with
```bash
start-newm -d
```
Other wayland compositors work just fine, such as Hyperland and Sway.

**Comment by pedrotrschneider** [comment]
@jbuchermn 
I'll try to answer all your questions in the order you asked for them:
- The wallpaper is mostly fine, except for four black squares in the four corners of the screen, each taking about 4% of the width of the screen (I'm on a laptop and can't test with other monitors at the moment, so that's the best I can do). I also can't show it to you, because no screenshot tool works.
- Tried to check if alacritty was running after pressing ```Mod-Enter``` but it seems to not be running (or at least ```ps ax``` doesn't see it, and ```killall``` says no process named alacritty is running.
- Disabling animations for the whole screen seems to not change anything.
- ```pywm-fullscreen -e firefox``` gives me a completely black screen. Tried running it with other programs as well, with the same result.
- As I mentioned above, I can't get any screenshot tool to work.
- The high cpu usage with ```newm-panel-basic``` is not an issue on my machine, probably a different issue related to something else. I tried disabling it all the same and nothing changed.

If there is anything else you want me to check, feel free to ask. I really want to make it work :)

**Comment by Cherro59** [comment]
Have also such problems.  My [log](https://pastebin.com/HLJ6jTBG)

**Comment by 0wg** [comment]
Also having the exact same issue.
